source/components/tabs/tab.ts

```diff
@@ -1,16 +1,36 @@
-import { Component, Optional, Inject, Input, Output, ViewChild, ContentChild, AfterViewInit, TemplateRef } from '@angular/core';
+import { Component, Optional, Inject, ElementRef, forwardRef, Input, Output, ViewChild, ContentChild, AfterViewInit, TemplateRef } from '@angular/core';
 import * as ng from 'angular';
 
-import { TabsetComponent, ITabHeader } from './tabset';
+import { TabsetComponent } from './tabset';
+import { FormComponent } from '../form/form';
 
+//TODO extend FormComponent
 @Component({
 	selector: 'rlTab',
-	template: require('./tab.html'),
+	template: require('./tab.html')
 })
 export class TabComponent {
-	header: ITabHeader;
+	@ContentChild(forwardRef(() => TabHeaderComponent))
+	header: TabHeaderComponent;
 
 	constructor() {
+	}
+}
+
+@Component({
+	selector: 'rlTabHeader',
+	template: '<ng-content></ng-content>',
+})
+export class TabHeaderComponent {

```

Comment:
OK let me take a look at those. It felt pretty dirty doing it that way but wasn't aware of the templates.